sdb/command.py

```diff
@@ -126,3 +126,46 @@ def call(self,
              objs: Iterable[drgn.Object]) -> Optional[Iterable[drgn.Object]]:
         # pylint: disable=missing-docstring
         raise NotImplementedError
+
+    def massage_input_and_call(self, objs: Iterable[drgn.Object]
+                              ) -> Optional[Iterable[drgn.Object]]:
+        """
+        Commands can declare that they accept input of type "foo_t*" by
+        setting their input_type. They can be passed input of type "void *"
+        or "foo_t" and this method will automatically convert the input
+        objects to the expected type (foo_t*).
+        """
```

Comment:
I really like this. Quick question: Did you think about the `avl_tree_t` <-> `struct avl_tree` matching in this context?

Reply:
That should "just work" once we merge #94 and I change the type equality operator on line 166 to use the new "canonical name" equality:
```
            if self.prog.pointer_type(first_obj_type) == expected_type:
```